oak_launcher_utils/src/launcher/virtualized.rs

```diff
@@ -29,60 +29,54 @@ use std::{
     process::Stdio,
 };
 
-fn path_exists(s: &str) -> Result<PathBuf, String> {
-    let path = PathBuf::from(s);
-    if !fs::metadata(s).map_err(|err| err.to_string())?.is_file() {
-        Err(String::from("path does not represent a file"))
-    } else {
-        Ok(path)
-    }
-}
-
 const PAGE_SIZE: usize = 4096;
 
-/// Parameters used for launching VM instances
+/// Represents parameters used for launching VM instances.
 #[derive(Parser, Clone, Debug, PartialEq)]
 pub struct Params {
     /// Path to the VMM binary to execute.
-    #[arg(long, value_parser = path_exists)]
+    #[arg(long, value_parser = Params::path_exists)]
     pub vmm_binary: PathBuf,
 
     /// Path to the enclave binary to load into the VM.
-    #[arg(long, value_parser = path_exists)]
+    #[arg(long, value_parser = Params::path_exists)]
     pub enclave_binary: PathBuf,
 
     /// Path to the Oak Functions application binary to be loaded into the enclave.
-    #[arg(long, value_parser = path_exists)]
+    #[arg(long, value_parser = Params::path_exists)]
     pub app_binary: PathBuf,
 
     /// Path to the BIOS image to use.
-    #[arg(long, value_parser = path_exists)]
+    #[arg(long, value_parser = Params::path_exists)]
     pub bios_binary: PathBuf,
 
     /// Port to use for debugging with gdb
     #[arg(long = "gdb")]
     pub gdb: Option<u16>,
 }
 
-/// Writes a chunk to a channel, and expects an acknowledgement containing the length of the chunk.
-fn write_chunk(channel: &mut dyn Channel, chunk: &[u8]) -> Result<()> {
-    channel.write(chunk)?;
-    let mut ack: [u8; 4] = Default::default();
-    channel.read(&mut ack)?;
-    if u32::from_le_bytes(ack) as usize != chunk.len() {
-        anyhow::bail!("ack wasn't of correct length");
+impl Params {
+    /// Checks if file with a given path exists.
+    fn path_exists(s: &str) -> Result<PathBuf, String> {
```

Comment:
This does not look like it should be a function on Params -- I think a top level stand-alone function seems more appropriate.

Reply:
Done.